scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -1641,6 +1681,40 @@ public Builder setNoServerSessionId(boolean flag) {
 			return this;
 		}
 
+		/**
+		 * Use anti replay filter.
+		 * 
+		 * @param enable {@code true} to enable filter. Default {@code true}.
+		 * @return this builder for command chaining.
+		 * @throws IllegalArgumentException if window filter is active.
+		 * @see http://tools.ietf.org/html/rfc6347#section-4.1
+		 */
+		public Builder setUseAntiReplayFilter(boolean enable) {
+			if (enable && Boolean.TRUE.equals(config.useWindowFilter)) {
+				throw new IllegalArgumentException("Window filter is active!");
+			}
+			config.useAntiReplayFilter = enable;
+			return this;
+		}
+
+		/**
+		 * Use window filter.
+		 * 
+		 * Messages too old for the filter window will pass the filter.
+		 * 
+		 * @param enable {@code true} to enable filter. Default {@code false}.
+		 * @return this builder for command chaining.
+		 * @throws IllegalArgumentException if anti replay window filter is active.
+		 * @see http://tools.ietf.org/html/rfc6347#section-4.1
+		 */
```

Comment:
Is it a normal use case to do that ? I feel this is more a kind of workaround?
If I am right, should we precise this is not really intended to be used ? (or something like this)

Bonus question, not directly linked to this PR, does the spec support windows size > 64 ? (min size seems to be 32) I didn't check the windows sliding algorithm to know if my question makes sense :grin:

Reply:
I guess, yes. But I don't see the benefit. Only 1 issue with that. Others seems not to have that issue. 64 is a long and more will make it more complex :-).